src/compare/vanilla.h

```diff
@@ -50,13 +51,60 @@ void VanillaCalcBondsIdx(const T *coords, std::size_t *idx, const T *box,
 
       r2 += rij * rij;
     }
-    *output++ = sqrt(r2);
+    *output++ = std::sqrt(r2);
+  }
+}
+
+template <typename T>
+void VanillaCalcAngles(const T *coords1, const T *coords2, const T *coords3,
+                       const T *box, unsigned int nvals, T *output) {
+  T rji[3];
+  T rjk[3];
+  T xp[3];
+  for (unsigned int i = 0; i < nvals; ++i) {
+    for (unsigned char j = 0; j < 3; ++j) {
+      T temp = coords1[i * 3 + j] - coords2[i * 3 + j];
+      T adj = std::round(temp / box[j]);
+      rji[j] -= adj * box[j];
+      temp = coords3[i * 3 + j] - coords2[i * 3 + j];
+      adj = std::round(temp / box[j]);
+      rjk[j] -= adj * box[j];
+    }
+    T x = rji[0] * rjk[0] + rji[1] * rjk[1] + rji[2] * rjk[2];
+
+    xp[0] = rji[1] * rjk[2] - rji[2] * rjk[1];
+    xp[1] = -rji[0] * rjk[2] + rji[2] * rjk[0];
+    xp[2] = rji[0] * rjk[1] - rji[1] * rjk[0];
+
+    T y = sqrt(xp[0] * xp[0] + xp[1] * xp[1] + xp[2] * xp[2]);
+
+    *output++ = atan2(y, x);
```

Comment:
Also I can't remember why we can't just use acos and a dot product here

Reply:
Cross product + arctan2 is meant to be more stable, but given how we are currently doing remainder i'm not sure it matters. I can implement an arccos + dot product version and we can compare?

 From some super quick benchmarks the atan2 op is slower, but that doesn't use vectors. Our final version will use vectorised trig, so atan2 and acos should event out.